### PR TITLE
VolumeGroup Snapshot 

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
Added changes for volume group snapshot

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Acceptance tests

### CODE REVIEW

File: examples/resources/powerstore_volumeGroupSnapshot/variables.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+variable "timeout" {
+  type=string
+  description = "Stores the timeout of PowerStore volume group snapshot."
+}
+
+variable "endpoint" {
+  type=string
+  description = "Stores the endpoint of PowerStore volume group snapshot. eg: https://10.1.1.1:443/api/rest/ where 443 is port where API requests are getting accepted"
```
Comment (prajwalpatil25) [inline examples/resources/powerstore_volumeGroupSnapshot/variables.tf]: description = "Stores the endpoint of PowerStore host. eg: https://10.1.1.1/api/rest"

File: examples/resources/powerstore_volumeGroupSnapshot/volumeGroupSnapshot_import.tf [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Below are the steps to import volume group snapshot :
+# Step 1 - To import a volume group snapshot , we need the id of that volume group snapshot
+# Step 2 - To check the id of the volume group snapshot we can make Get request to volume group snapshot endpoint. eg. https://10.0.0.1/api/rest/volume_group --header 'type: Snapshot' which will return list of all volume group snapshots ids.
+# Step 3 - Add empty resource block in tf file.
+# eg.
+ resource "powerstore_volumeGroupSnapshot" "resource_block_name" {
```
Comment (prajwalpatil25) [inline examples/resources/powerstore_volumeGroupSnapshot/volumeGroupSnapshot_import.tf]: the resource block should be commented.

File: examples/resources/powerstore_volumeGroupSnapshot/volumeGroupSnapshot.tf [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Commands to run this tf file : terraform init && terraform plan && terraform apply
+# Create, Update, Delete is supported for this resource
+# To import , check volumeGroupSnapshot_import.tf for more info
+# name, volume_group_id/volume_group_name and expiration_timestamp are the required attributes to create and update
+# description is the optional attributes
```
Comment (shenda1) [inline examples/resources/powerstore_volumeGroupSnapshot/volumeGroupSnapshot.tf]: attribute

File: powerstore/resource_volume_group_snapshot.go [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+			"name": schema.StringAttribute{
+				Required:            true,
+				Description:         "Name of the volume group snapshot.",
+				MarkdownDescription: "Name of the volume group snapshot.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"volume_group_id": schema.StringAttribute{
+				Optional:            true,
```
Comment (shenda1) [inline powerstore/resource_volume_group_snapshot.go:54]: Can volume_group_id added as computed attribute? same for name as well?

Comment (Sakshi-dell) [reply powerstore/resource_volume_group_snapshot.go:54]: As discussed, volume group id has been made computed, but not volume group name

**Comment by Sakshi-dell** [comment]
image
Acceptance test current coverage

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**